### ISSUE TITLE
set default value if null (or meet some condition) in deserialization

### ISSUE THREAD

**Issue by wszk1992** [FasterXML/jackson-future-ideas#39]
I'm wondering if we can have an annotation (field level and class level) to set a default value if the corresponding JSON value is null in deserialization.

```json
{
  "name": "Jack"
}
````

```java
class People {
  String name;
  @JsonDefault(intDefault = 10)
  Integer age;
}
```
After deserialization, We can have people.name == "Jack" and people.age == 10.

```java
@JsonDefault(floatDefault = 0.0f, intDefault = 0)
class People {
  String name;
  Integer age;
  Float weight;
}
```
After deserialization, We can have people.name == "Jack", people.age == 0 and people.weight == 0.0

**Comment by cowtowncoder** [comment]
Since Jackson 2.9, there is `@JsonSetter` which allows for defaulting to "empty", as well as failing (throwing exception) or skipping (leaving default value). This is explained here:

https://medium.com/@cowtowncoder/jackson-2-9-features-b2a19029e9ff

and probably covers some of the use cases.

This does not quite cover cases that might occur here but I mention it in any case anyone is looking for existing functionality.

Other than that, we have no current plans to extend value handling towards configurable defaults: the main approach is to expect POJOs to be constructed with defaults -- this coupled with `Nulls.SKIP` might work for most use cases.

**Comment by MartinX3** [comment]
It would be nice, if my kotlin project would deserialize as example null strings to empty strings `""` and send it to the frontend.

**Comment by gabrieljones** [comment]
Can we have `@JsonSetter(missing=Missing.AS_EMPTY)`?

**Comment by cowtowncoder** [comment]
Yes, if this feature was implemented one possibility could be to use constant like that.

**Comment by swayamraina** [comment]
Are we thinking of doing a `@JsonDefault` at a field level?

**Comment by cowtowncoder** [comment]
@swayamraina I think this is what is requested.
